agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -279,42 +279,79 @@ private void log(Logger logger) {
         }
     }
 
-    public static Configuration loadJsonConfigFile(Path configPath) throws IOException{
-        if (!Files.exists(configPath)) {
-            throw new IllegalStateException("config file does not exist: " + configPath);
-        }
-        
-        BasicFileAttributes attributes = Files.readAttributes(configPath, BasicFileAttributes.class);
-        // important to read last modified before reading the file, to prevent possible race condition
-        // where file is updated after reading it but before reading last modified, and then since
-        // last modified doesn't change after that, the new updated file will not be read afterwards
-        long lastModifiedTime = attributes.lastModifiedTime().toMillis();
+    static Configuration getConfigurationFromConfigFile(Path configPath, boolean strict) throws IOException {
         try (InputStream in = Files.newInputStream(configPath)) {
             Moshi moshi = MoshiBuilderFactory.createBuilderWithAdaptor();
-            JsonAdapter<Configuration> jsonAdapter = moshi.adapter(Configuration.class).failOnUnknown();
+            JsonAdapter<Configuration> jsonAdapter = strict ? moshi.adapter(Configuration.class).failOnUnknown() :
+                    moshi.adapter(Configuration.class);
             Buffer buffer = new Buffer();
             buffer.readFrom(in);
             try {
                 Configuration configuration = jsonAdapter.fromJson(buffer);
-                if (configuration.instrumentationSettings != null) {
-                    throw new IllegalStateException("It looks like you are using an old applicationinsights.json file" +
-                            " which still has \"instrumentationSettings\", please see the docs for the new format:" +
-                            " https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-standalone-config");
-                }
-                configuration.configPath = configPath;
-                configuration.lastModifiedTime = lastModifiedTime;
                 return configuration;
             } catch(JsonDataException ex) {
-                throw new FriendlyException("Application Insights Java agent's configuration file "+configPath.toAbsolutePath().toString()+" has the following issue:\n"+ex.getMessage(),
-                        "Learn more about configuration options here: https://go.microsoft.com/fwlink/?linkid=2153358");
-
+                Logger logger = LoggerFactory.getLogger(ConfigurationBuilder.class);
+                logger.warn(ex.getMessage());
```

Comment:
I think this should be inside the "if strict" condition below

Also, good to make Logger `private static final` at top of class